### DISCUSSION TITLE
`outline: none` support

### DISCUSSION THREAD

**Discussion by ShayanTheNerd** [tailwindlabs/tailwindcss#12768] [Ideas]
There's already this utility called `outline-none`, but it doesn't quite act as the name suggests. We'd expect it to generate `outline: none` or `outline-style: none`  CSS style, but instead, it's adding a 2px, solid, transparent outline with an offset of 2px. TBH, I'm not really sure why this is how it is. What I do know, though, is that when I see a utility called `outline-none`, I'd expect it to _actually_ remove the outline. A real-world example could be when you want to get rid of Firefox's not-so-beautiful outline and implement a custom one.

I'd like to know about the reason behind this behavior, in addition to having the utility do exactly as the name says; so that we won't have to write something like `outline: none` in Vanilla CSS.

**Comment by wongjn** [comment]
For the reason that `outline-none` applies `outline: 2px solid transparent`, we can refer to [the documentation](https://tailwindcss.com/docs/outline-style#removing-outlines):

> The `outline-none` utility is implemented using a transparent outline under the hood to ensure elements are still visibly focused to [Windows high contrast mode](https://blogs.windows.com/msedgedev/2020/09/17/styling-for-windows-high-contrast-with-new-standards-for-forced-colors/) users.

For what it's worth, you could use an [arbitrary property class](https://tailwindcss.com/docs/adding-custom-styles#arbitrary-properties) to apply `outline: none` with `[outline:none]`.
